sklearnex/tests/test_patching.py

```diff
@@ -14,107 +14,236 @@
-    return result
 
+        _ = pairwise_distances(X.reshape(1, -1), metric=metric)
+    assert all(
+        [
+            "running accelerated version" in i.message
+            or "fallback to original Scikit-learn" in i.message
+            for i in caplog.records
+        ]
+    ), f"sklearnex patching issue in pairwise_distances with log: \n{caplog.text}"
 
-result_log = get_result_log()
 
+@pytest.mark.parametrize(
+    "dtype", [i for i in DTYPES if "32" in i.__name__ or "64" in i.__name__]
+)
+@pytest.mark.parametrize(
+    "dataframe, queue", get_dataframes_and_queues(dataframe_filter_="numpy")
+)
+def test_roc_auc_score_patching(caplog, dataframe, queue, dtype):
+    if dtype in DTYPES[-2:]:
+        pytest.skip("Windows issue with unsigned ints")
+    with caplog.at_level(logging.WARNING, logger="sklearnex"):
+        rng = nprnd.default_rng()
+        X = _convert_to_dataframe(
+            rng.integers(2, size=1000),
+            sycl_queue=queue,
+            target_df=dataframe,
+            dtype=dtype,
+        )
+        y = _convert_to_dataframe(
+            rng.integers(2, size=1000),
+            sycl_queue=queue,
+            target_df=dataframe,
+            dtype=dtype,
+        )
 
-@pytest.mark.parametrize("configuration", result_log)
-def test_patching(configuration):
-    if "OPT" in result_log[configuration]:
-        return
-    for skip in TO_SKIP:
-        if re.search(skip, configuration) is not None:
-            pytest.skip("SKIPPED", allow_module_level=False)
-    raise ValueError("Test patching failed: " + configuration)
+        _ = roc_auc_score(X, y)
+    assert all(
+        [
+            "running accelerated version" in i.message
+            or "fallback to original Scikit-learn" in i.message
+            for i in caplog.records
+        ]
+    ), f"sklearnex patching issue in roc_auc_score with log: \n{caplog.text}"
 
 
-def _load_all_models(patched):
-    if patched:
-        patch_sklearn()
+@pytest.mark.parametrize("dtype", DTYPES)
+@pytest.mark.parametrize(
+    "dataframe, queue", get_dataframes_and_queues(dataframe_filter_="numpy")
+)
+@pytest.mark.parametrize("estimator, method", gen_models_info(PATCHED_MODELS))
+def test_standard_estimator_patching(caplog, dataframe, queue, dtype, estimator, method):
+    with caplog.at_level(logging.WARNING, logger="sklearnex"):
+        est = PATCHED_MODELS[estimator]()
 
-    models = {}
-    for patch_infos in get_patch_map().values():
-        maybe_class = getattr(patch_infos[0][0][0], patch_infos[0][0][1], None)
-        if (
-            maybe_class is not None
-            and isclass(maybe_class)
-            and issubclass(maybe_class, BaseEstimator)
+        if estimator == "TSNE" and method == "fit_transform":
+            pytest.skip("TSNE.fit_transform is too slow for common testing")
+        elif (
+            estimator == "Ridge"
+            and method in ["predict", "score"]
+            and dtype in DTYPES[-2:]
```

Comment:
```suggestion
            and dtype in set(DTYPES) - {"np.int8", "np.int16"}
```

You should not assume that `DTYPES` is in any particular order.
Also, if it only fails on Windows, you should only skip it on Windows.

Reply:
totally correct